packages/next-common/components/hybridRender.js

```diff
@@ -0,0 +1,25 @@
+import React, { useState, useEffect } from "react";
+
+export default function HybridRender({ children }) {
+  const [isMounted, setIsMounted] = useState(false);
+
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
+
+  return <div key={isMounted ? "client" : "server"}>{children}</div>;
+}
+
+export function ClientOnly({ children, fallback = null }) {
```

Comment:
We have a predefined constant IS_SERVER, which is a better approach

export const IS_SERVER = typeof window === "undefined";